### PR TITLE
DEVOPS-3254: fix(deployment scheduler): use apps/v1 instead of extenstion/v1beta1 for deployments

### PR DESCRIPTION
**Pull request by ruthwells**
And fix rbac api version.

Tested in minikube - brings casebook web/worker up and down fine.

### CODE REVIEW

File: schedule_scaling/schedule_scaling.py [hunk 1]
```diff
@@ -19,6 +20,8 @@ def create_job_directory():
     if os.path.isdir(temp__dir):
         shutil.rmtree(temp__dir)
     pathlib.Path(temp__dir).mkdir(parents=True, exist_ok=True)
+    # need to copy cadeployment.py to override pykube.Deployment with the correct api version
```
Comment (zhelyan) [inline schedule_scaling/schedule_scaling.py]: why does it need to be copied over to the temp dir? The file is already in the project

Comment (ruthwells) [reply schedule_scaling/schedule_scaling.py]: @zhelyan, yes, that's what I thought. But until I copied it across, the jobs running from the temp dir were failing: 
`Traceback (most recent call last):
  File "/tmp/scaling_jobs/casebook-worker-0.py", line 5, in <module>
    import cadeployment
ImportError: No module named 'cadeployment'
Traceback (most recent call last):
  File "/tmp/scaling_jobs/casebook-web-0.py", line 5, in <module>
    import cadeployment
ImportError: No module named 'cadeployment'
`
Copying the file across solved the problem (but maybe not in the correct way...)

File: Dockerfile [hunk 2]
```diff
@@ -25,6 +25,8 @@ RUN sudo apt-get update \
 ADD schedule_scaling /root/schedule_scaling
 COPY ./run_missed_jobs.py /root
 RUN chmod a+x /root/run_missed_jobs.py
+# allow us to override pykube.Deployment with the correct api version
+COPY ./cadeployment.py /root/schedule_scaling
```
Comment (michelesr) [inline Dockerfile]: don't call it cadeployment please... this is a public repository and the code is generic enough than can be used by other organizations

File: cadeployment.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from pykube import Deployment
+import pykube
+
+# extends pykube.Deployment, overriding version "v1beta1" with "apps/v1"
+class CADeployment(pykube.Deployment):
```
Comment (michelesr) [inline cadeployment.py]: same as above

File: schedule_scaling/schedule_scaling.py [hunk 4]
```diff
@@ -8,6 +8,7 @@
 import re
 import urllib.request
 import boto3
+import cadeployment
```
Comment (michelesr) [inline schedule_scaling/schedule_scaling.py]: same as above

File: schedule_scaling/schedule_scaling.py (hunk 1 above)
Comment (michelesr) [reply schedule_scaling/schedule_scaling.py]: this can be fixed by setting the `PYTHONPATH` environment variable to contain `/root/shedule_scaling`

**Review by michelesr — CHANGES_REQUESTED** [review]
Please use a generic name for the custom deployment class and module (not CA), and set `PYTHONPATH` to make the module available without copying the file on the same directory.

https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH

File: cadeployment.py (hunk 3 above)
Comment (michelesr) [reply cadeployment.py]: you can simply call the class `Deployment`, as is hosted in a different module so won't clash with pykube, but if you think that might cause confusion that call it `CustomDeployment` or `AppsV1Deployment`

File: cadeployment.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from pykube import Deployment
+import pykube
+
+# extends pykube.Deployment, overriding version "v1beta1" with "apps/v1"
```
Comment (michelesr) [inline cadeployment.py]: the comment here should say overriding version `extensions/v1beta1` with `apps/v1`

**Review by zhelyan — APPROVED** [review]

**Review by michelesr — APPROVED** [review]

**Pull request merged**